src/components/fields/location-field/location-modal/location-modal.tsx

```diff
@@ -97,6 +97,12 @@ const LocationModal = ({
 	useEffect(() => {
 		if (!window) return;
 
+		// If formValues persist upon re-entry into the location modal,
+		// then set selectedAddressInfo accordingly.
+		if (formValues) {
+			setSelectedAddressInfo(formValues);
+		}
+
```

Comment:
what's the rationale of setting the formValues when it will be set at: https://github.com/LifeSG/web-frontend-engine/blob/MOL-13392-fix/src/components/fields/location-field/location-modal/location-search/location-search.tsx#L171

Reply:
Yes, you are right. I have removed this section of code.